### PR TITLE
DPE command validation and DeriveChild changes

### PR DESCRIPTION
**Pull request by hpya93**
This contains-

_•Test all failure cases defined in the DPE iRoT Profile_

- Added a failure check in the initialize context and getcertificatechain command. Other fail checks are already present in the code which I have tested.

_•For all input enums/flags for all DPE commands, ensure there is a test case for each permutation._

_•Add tests to ensure localities are properly isolated from each other for all commands_

- I have done testing on separate localities in simulator side (TestInitializeContext using multiple localities)

- The code for separating localities is already there in the command file. Just checked that the initialize context is called for each locality.

- In emulator side, we are not setting any locality to the DPE instance when executing commands.

_•Validate proper behavior to against the feature support matrix (exposed by GetProfile)_

- Validated the getprofile modes for both simulator and emulator using code from this 

_•DeriveChild  - Test limits on number of DeriveChild calls_
- Have checked whether proper error is reported when number of derived child exceed MAX TCI nodes

_•Test that DeriveChild privileges (like ALLOW_CA) are correctly bounded and that a DPE context cannot escalate its privileges_
 - Multi Tcb validation for matching TCITYPE passed to Derive child request   
 - Have checked whether the proper error is returned when feature flags are set in derived child context but not supported by parent context  

**Query:**

In fail cases of getcertificatechain command, spec says 'If OFFSET is larger than the full certificate chain, this API shall return DPE_STATUS_INVALID_ARGUMENT'

But,

The full cert chain size is computed in runtime, by repeated calls to GetCertificateChain()

So, we can't check if offset > fullcertchainsize

This can be checked only if we pass a negative test input like

cases := []GetCertificateReq{
{size: 2048, offset: 100000},
{size: 2048, offset: 100000}, // Negative case.
}

So, should we write a negative test case like this?

### CODE REVIEW

File: verification/client.go [hunk 1]
```diff
@@ -236,6 +239,18 @@ func (c *Client[_, _]) GetCertificateChain() (*GetCertificateChainResp, error) {
 	return &certs, nil
 }
 
+// DeriveChild() calls the DPE CertifyKey command.
```
Comment (sree-revoori1) [inline verification/client.go:242]: *calls the DPE DeriveChild command

File: verification/deriveChild_test.go [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+			log.Fatal(err)
+		}
+		defer d.PowerOff()
+	}
+	client, err := NewClient256(d)
+	if err != nil {
+		t.Fatalf("Could not initialize client: %v", err)
+	}
+
+	tciNodeCount := uint32(1)
```
Comment (sree-revoori1) [inline verification/deriveChild_test.go:37]: can you add a comment here that the initial tciNodeCount is 1 due to auto-init?

File: verification/client.go (hunk 1 above)
Comment (hpya93) [reply verification/client.go:242]: Hi @sree-revoori1  , seems the code flow is modified few days back and this tree needs significant changes to incorporate the new code. I shall commit the changes after rebasing the code. 

Thank you!

**Comment by hpya93** [comment]
Have incorporated latest pattern changes in PR

**Comment by hpya93** [comment]
@sree-revoori1 Can you please review and let us know the comments on this PR?

**Pull request closed without merging**